corehq/apps/hqwebapp/forms.py

```diff
@@ -111,34 +107,23 @@ def crispy_form_fields(self, context):
 
 
 class AppTranslationsBulkUploadForm(BulkUploadForm):
+    language = forms.CharField(widget=forms.HiddenInput)
     validate = forms.BooleanField(label="Just validate and not update translations", required=False,
                                   initial=False)
-    language = forms.ChoiceField(label=_("Language to validate"), choices=(), required=False)
 
     def crispy_form_fields(self, context):
         crispy_form_fields = super(AppTranslationsBulkUploadForm, self).crispy_form_fields(context)
+        if context.get('can_select_language'):
+            crispy_form_fields.extend([
+                InlineField('language', data_bind="value: lang")
+            ])
         if context.get('can_validate_app_translations'):
             crispy_form_fields.extend([
-                InlineField('validate', data_bind="checked: validate"),
                 crispy.Div(
-                    crispy.Field('language'),
-                    css_class='col-sm-2',
-                    data_bind="visible: validate"
-                ),
+                    InlineField('validate'),
+                    data_bind="visible: lang",
+                )
             ])
-            self.fields['language'].choices = tuple_of_copies(self.app.langs, blank=False)
-            self.fields['language'].initial = self.app.default_language
```

Comment:
Looks like the choices is still all app langs but the initial value is "All" instead of the default language as set up for the validation feature. Would need to address that for the validation feature, I can take that in a follow up PR once this is merged.

Reply:
The UI does now default to "All" and hides the validation checkbox. The validation shows up when you pick a language: http://g.recordit.co/3VbpJ6Z4nG.gif

I'm not totally in love with this UX, but we shouldn't have more than one language dropdown in the same form. If this UI is confusing, we should look at separating validation from uploading, rather than having validation be a checkbox.

Reply:
Interesting work around there. Would just need to make people aware of this change, i am sure they would be searching for the validation option.